### PR TITLE
conflicts: make `update_from_content()` work with only `FileId`s

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

**Review by yuja — APPROVED** [review]

**Comment by martinvonz** [comment]
Sorry, i just added two more commits to this PR. Did you see those before you approved?

File: lib/src/conflicts.rs [hunk 1]
```diff
@@ -300,7 +300,7 @@ pub fn update_from_content(
     store: &Store,
     path: &RepoPath,
     content: &[u8],
-) -> BackendResult<Option<Merge<Option<FileId>>>> {
+) -> BackendResult<Merge<Option<FileId>>> {
```
Comment (yuja) [inline lib/src/conflicts.rs:303]: nit: needs to update the doc comment

**Review by yuja — APPROVED** [review]
> Did you see those before you approved?

Nah, but they all look good.

File: lib/src/conflicts.rs (hunk 1 above)
Comment (martinvonz) [reply lib/src/conflicts.rs:305]: Done.

Comment (ilyagr) [reply lib/src/conflicts.rs:305]: I'm so happy to see this diff :)

**Pull request merged**